modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -39,12 +59,24 @@ function getValidFields(properties: any): Array<IField> {
 			return;
 		}
 
-		if (propertyValue.$ref) {
+		if (type === 'object' && propertyValue.$ref) {
```

Comment:
I was hoping that this solution will cover any complex objects, but that cannot be the case. This PR will only cover Objects relationships, and their have very specific structure that we need to parse.